src/components/shared/TaxDonationModal.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import { useState } from "react";
+import dynamic from "next/dynamic";
+import Image from "next/image";
+
+import { DonationInstructions } from "../Support/DonationInstructions";
+import { Button } from "./Button/Button";
+import opp from "../../../public/images/support/opp.png";
+import pit from "../../../public/images/support/pit.png";
+
+const Modal = dynamic(() => import("../Modal/Modal").then((mod) => mod.Modal), {
+  ssr: false,
+});
+
+export const TaxDonationModal = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        label="Przekaż podatek"
+        ariaDescription="Przekaż podatek"
+        onClick={() => setIsModalOpen(true)}
+      />
+      {isModalOpen && (
+        <Modal isModalOpen={isModalOpen} setIsModalOpen={setIsModalOpen}>
```

Comment:
Having dedicated `open` & `close` props & methods would be preferred for me. But I know it is due to the props of the already existing `Modal` component. Maybe we could adjust that?

Reply:
Also to simplify boolean setting to true/false and toggling it we could introduce some custom hook to simplify the usage.

Lookup `useBoolean` or `useDisclosure` hook in popular solutions (google it 😄)